pkg/services/sqlstore/permissions/dashboards_bench_test.go

```diff
@@ -0,0 +1,143 @@
+		filter := permissions.NewAccessControlDashboardPermissionFilter(usr, models.PERMISSION_VIEW, "")
+		var result int
+		err := store.WithDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
+			q, params := filter.Where()
+			_, err := sess.SQL("SELECT COUNT(*) FROM dashboard WHERE "+q, params...).Get(&result)
+			return err
+		})
+		require.NoError(b, err)
+		assert.Equal(b, numDashboards, result)
+	}
+}
+
+func setupBenchMark(b *testing.B, numUsers, numDashboards int) db.DB {
+	store := db.InitTestDB(b)
+	now := time.Now()
+	err := store.WithDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
+		dashes := make([]models.Dashboard, 0, numDashboards)
+		for i := 1; i <= numDashboards; i++ {
+			str := strconv.Itoa(i)
+			dashes = append(dashes, models.Dashboard{
+				OrgId:    1,
+				IsFolder: false,
+				Uid:      str,
+				Slug:     str,
+				Title:    str,
+				Data:     simplejson.New(),
+				Created:  now,
+				Updated:  now,
+			})
+		}
+
+		err := batch(len(dashes), 1000, func(start, end int) error {
+			_, err := sess.InsertMulti(dashes[start:end])
+			return err
+		})
+		require.NoError(b, err)
+
+		roles := make([]accesscontrol.Role, 0, numUsers)
+		assignments := make([]accesscontrol.UserRole, 0, numUsers)
+		permissions := make([]accesscontrol.Permission, 0, numUsers*numDashboards)
+		for i := 1; i <= numUsers; i++ {
+			name := fmt.Sprintf("managed_%d", i)
+			roles = append(roles, accesscontrol.Role{
+				UID:     name,
+				Name:    name,
+				Updated: now,
+				Created: now,
+			})
+			assignments = append(assignments, accesscontrol.UserRole{
+				RoleID:  int64(i),
+				UserID:  int64(i),
+				Created: now,
+			})
+			for _, dash := range dashes {
+				permissions = append(permissions, accesscontrol.Permission{
+					RoleID:  int64(i),
+					Action:  dashboards.ActionDashboardsRead,
+					Scope:   dashboards.ScopeDashboardsProvider.GetResourceScopeUID(dash.Uid),
+					Updated: now,
+					Created: now,
+				})
+			}
+		}
+
+		err = batch(len(roles), 5000, func(start, end int) error {
+			_, err := sess.InsertMulti(roles[start:end])
+			return err
+		})
+		require.NoError(b, err)
+
+		err = batch(len(assignments), 5000, func(start, end int) error {
+			_, err := sess.InsertMulti(assignments[start:end])
+			return err
+		})
+		require.NoError(b, err)
+
+		err = batch(len(permissions), 5000, func(start, end int) error {
+			_, err := sess.InsertMulti(permissions[start:end])
+			return err
+		})
+		require.NoError(b, err)
+		return nil
+	})
+
+	require.NoError(b, err)
+	return store
+}
+
+func BenchmarkDashboardPermissionFilter_100_100(b *testing.B) {
+	benchmarkDashboardPermissionFilter(b, 100, 100)
+}
+
+func BenchmarkDashboardPermissionFilter_100_1000(b *testing.B) {
+	benchmarkDashboardPermissionFilter(b, 100, 1000)
+}
+
+func BenchmarkDashboardPermissionFilter_300_10000(b *testing.B) {
```

Comment:
could you post the results from the benchmark on the PR so we have a basis we can come back to?